src/model/backup.ts

```diff
@@ -232,7 +246,40 @@ async function restoreCurrentBackupIntoKeychain(
         if (endsWith(key, seedPhraseKey) || endsWith(key, privateKeyKey)) {
           accessControl = privateAccessControlOptions;
         }
-        if (typeof value === 'string') {
+
+        const hasBiometricsEnabled = await getSupportedBiometryType();
+
+        if (!hasBiometricsEnabled && android && endsWith(key, seedPhraseKey)) {
+          const parsedValue = JSON.parse(value);
+          const { seedphrase } = parsedValue;
+
+          const isBackupWasSavedWithPIN =
+            seedphrase?.includes('salt') && seedphrase?.includes('cipher');
+
+          if (!isBackupWasSavedWithPIN) {
+            //interrupt loader
+            doItIfStartPINCreation();
+            const userPIN = await saveNewAuthenticatePIN();
+            doItIfFinishPINCreation();
+
+            const encryptedSeed = await encryptor.encrypt(userPIN, seedphrase);
+
+            if (encryptedSeed) {
+              // as the wallet was backuped with faceId in `seedphrase` we have
+              // a seed(as a words), so we need to store this seed with PIN
+              // and put that encrypted data instead this seed
```

Comment:
I posted fixes to couple typos above, but I don't understand the core idea of this sentence.

The wallet was backed up with face ID. And because of that, in `seedphrase` (the parsed variable, right?) we have the seed phrase as a string/array of words.

And because of that, we need to store this seed with PIN... **Why?**

...and put that encrypted data instead of this seed. **Which encrypted data? Put where? Instead of which seed, the one we need to store with PIN?**

Reply:
I shortened this comment and removed the extraneous content